examples/pyg/graph_sage_unsup_quiver.py

```diff
@@ -0,0 +1,145 @@
+######################
+import quiver
+from quiver.pyg import GraphSageSampler
+
+EPS = 1e-15
+
+dataset = 'Cora'
+# path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', dataset)
+path = '/data/products/' + dataset
+dataset = Planetoid(path, dataset, transform=T.NormalizeFeatures())
+data = dataset[0]
+
+train_idx = data.train_mask.nonzero(as_tuple=False).view(-1)
+
+#############################
+# Original Pyg Code
+#############################
+# train_loader = NeighborSampler(data.edge_index, sizes=[10, 10], batch_size=256,
+#                                shuffle=True, num_nodes=data.num_nodes)
+train_loader = torch.utils.data.DataLoader(train_idx,
+                                           batch_size=256,
+                                           shuffle=True,
+                                           drop_last=True)
+
+csr_topo = quiver.CSRTopo(data.edge_index)
+
+quiver_sampler = GraphSageSampler(csr_topo, sizes =[10, 10], device=0)
+
+def sample(edge_index, batch):
+    batch = torch.tensor(batch)
+    row, col = edge_index[0], edge_index[1]
+
+    # For each node in `batch`, we sample a direct neighbor (as positive
+    # example) and a random node (as negative example):
+    pos_batch = random_walk(row, col, batch, walk_length=1,
+                            coalesced=False)[:, 1]
+
+    neg_batch = torch.randint(0, self.adj_t.size(1), (batch.numel(), ),
+                                dtype=torch.long)
+
+    batch = torch.cat([batch, pos_batch, neg_batch], dim=0)
+    return quiver_sampler.sample(batch)
+
+
+class SAGE(nn.Module):
+    def __init__(self, in_channels, hidden_channels, num_layers):
+        super(SAGE, self).__init__()
+        self.num_layers = num_layers
+        self.convs = nn.ModuleList()
+        for i in range(num_layers):
+            in_channels = in_channels if i == 0 else hidden_channels
+            self.convs.append(SAGEConv(in_channels, hidden_channels))
+
+    def forward(self, x, adjs):
+        for i, (edge_index, _, size) in enumerate(adjs):
+            x_target = x[:size[1]]  # Target nodes are always placed first.
+            x = self.convs[i]((x, x_target), edge_index)
+            if i != self.num_layers - 1:
+                x = x.relu()
+                x = F.dropout(x, p=0.5, training=self.training)
+        return x
+
+    def full_forward(self, x, edge_index):
+        for i, conv in enumerate(self.convs):
+            x = conv(x, edge_index)
+            if i != self.num_layers - 1:
+                x = x.relu()
+                x = F.dropout(x, p=0.5, training=self.training)
+        return x
+
+
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+model = SAGE(data.num_node_features, hidden_channels=64, num_layers=2)
+model = model.to(device)
+optimizer = torch.optim.Adam(model.parameters(), lr=0.01)
+####################
+# Original Pyg Code
+####################
+x, edge_index = data.x.to(device), data.edge_index.to(device)
```

Comment:
remove `x`, only use `quiver.feature`, but you need to reinstall `torch-scatter` first